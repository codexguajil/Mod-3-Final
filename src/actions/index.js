export const storeCards = (cards) => ({
  type: 'ADD_ALL',
  cards
})