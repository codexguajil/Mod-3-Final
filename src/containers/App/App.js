import React, { Component } from 'react';
import logo from './logo.svg';
import { connect } from 'react-redux';
import {storeCards} from '../../actions';
import {fetchAllCards} from '../../helpers/apiCalls';
import {houseCleaner} from '../../helpers/dataCleaner';
import CardsContainer from '../CardsContainer/CardsContainer';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      isLoading: false,
    }
  }

  componentDidMount = async () => {
    this.setState({isLoading: true})
    await this.fetchCards()
    this.setState({isLoading: false})
  }

  fetchCards = async () => {
    const cards = await fetchAllCards()
    const cleanCards = houseCleaner(cards)
    console.log(cleanCards)
    this.props.storeCards(cleanCards)
  }

  render() {
    const {isLoading} = this.state
    return (
      <div className='App'>
        <div className='App-header'>
        {isLoading &&
          <div id="loading-gif"></div>
        }
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westerosss</h2>
        </div>
        <CardsContainer />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  cards: state.cards
})

export const mapDispatchToProps = (dispatch) => ({
  storeCards: (cards) => dispatch(storeCards(cards))
})

export default connect(null, mapDispatchToProps)(App);
