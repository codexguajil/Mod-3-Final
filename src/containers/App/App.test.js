import React from 'react'
import { shallow } from 'enzyme'
import {App, mapDispatchToProps, mapStateToProps} from './App'
import {storeCards} from '../../actions'

describe('App', () => {
  let wrapper;
  
  it('matches the snapshot', () => {
    wrapper = shallow(
      <App />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it.skip('should set is loading to false after fetching the cards', async() => {

    await wrapper.componentDidMount()
    expect(wrapper.state('isLoading')).toEqual(false)
  })

  it('should return an object with the cards store array', () => {
    const mockState = {
      cards: [
        {},
        {},
        {}
      ],
      id: 3
    }

    const expected = {cards: [
        {},
        {},
        {}
    ]}

    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })

  it('calls dispatch with with the storeCards action when fetchCards is called', () => {
    const mockDispatch = jest.fn()
    const actionToDispatch = storeCards([{}, {}, {}])

    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.storeCards([{}, {}, {}])

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})