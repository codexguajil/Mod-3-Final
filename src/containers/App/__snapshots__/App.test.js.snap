// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
        >
          <div
            id="loading-gif"
          />
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />
          <h2>
            Welcome to Westerosss
          </h2>
        </div>,
        <Connect(CardsContainer) />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="loading-gif"
            />,
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />,
            <h2>
              Welcome to Westerosss
            </h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "loading-gif",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westerosss",
            },
            "ref": null,
            "rendered": "Welcome to Westerosss",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
          >
            <div
              id="loading-gif"
            />
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
            <h2>
              Welcome to Westerosss
            </h2>
          </div>,
          <Connect(CardsContainer) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="loading-gif"
              />,
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
              />,
              <h2>
                Welcome to Westerosss
              </h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "loading-gif",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westerosss",
              },
              "ref": null,
              "rendered": "Welcome to Westerosss",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
